### ISSUE TITLE
1.1.5 publish through redis fail

### ISSUE THREAD

**Issue by mapledan** [slact/nchan#354]
I got this error log when pub the message.
```
/opt/nchan-1.1.5/src/util/nchan_msg.c:94: msg_reserve: Assertion `msg->storage != NCHAN_MSG_SHARED' failed.
2017/05/09 10:13:15 [alert] 6488#0: worker process 6493 exited on signal 6 (core dumped)
```

My nginx config
```

user  www-data;
worker_processes  16;

events {
    worker_connections  1024;
    accept_mutex on;
}


http {
    include       mime.types;
    default_type  application/octet-stream;

    sendfile        on;

    upstream redis_cluster {
        nchan_redis_server redis://127.0.0.1:7000;
        nchan_redis_server redis://127.0.0.1:7001;
        nchan_redis_server redis://127.0.0.1:7002;
    }

    keepalive_timeout  65s;
    send_timeout 65s;
    client_max_body_size 5m;
    client_body_timeout 65s; # or 60s
    client_header_timeout 65s;
    proxy_connect_timeout 65s;
    proxy_read_timeout 65s;
    nchan_message_buffer_length 5000;
    nchan_message_timeout 24h;

    gzip  on;
    gzip_types application/json;

    server {
        listen 80;
        listen 443 ssl http2;
        ssl_certificate      /usr/local/nginx/ssl/nginx.pem;
        ssl_certificate_key  /usr/local/nginx/ssl/nginx.key;
        ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
        ssl_ecdh_curve secp384r1;
        ssl_prefer_server_ciphers on;
        ssl_session_cache shared:SSL:10m;
        ssl_session_tickets off;
        ssl_ciphers ECDHE+AESGCM:ECDHE+AES:RSA+AES:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!DSS:!3DES:!PSK:!kEDH;

        server_name  localhost;
        server_tokens off;

        root /var/www/html; 

        location ~ /sub/(\w+) {
            nchan_channel_id $2;
            nchan_subscriber longpoll;
            nchan_authorize_request /api/$1/auth.php;
            nchan_redis_pass redis_cluster;
            nchan_longpoll_multipart_response raw;
            nchan_subscriber_compound_etag_message_id on;
        }

        location ~ /pub/internal/(\w+)$ {
            nchan_publisher http;
            nchan_redis_pass redis_cluster;
            nchan_channel_id $1;
            allow 127.0.0.0/8;
            deny all;
        }

        location /nchan {
          nchan_stub_status;
          allow 127.0.0.0/8;
          allow 172.30.0.0/16;
          allow 172.25.0.0/16;
          deny all;
        }

        location ~ \.php {
            fastcgi_split_path_info ^(.+\.php)(/.+)$;
            include fastcgi_params;
            fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
            fastcgi_param SCRIPT_NAME $fastcgi_script_name;
            fastcgi_index index.php;
            fastcgi_pass unix:/run/php/php7.0-fpm.sock;
       }

        error_page   500 502 503 504  /50x.html;
        location = /50x.html {
            root   html;
        }

        error_page  403 404  /40x.html;
        location = /40x.html {
            root   html;
        }

        location = /favicon.ico {
          log_not_found off;
        }
    }
}

```

**Comment by slact** [comment]
Are you able to reproduce this error? Do you by any chance have a coredump I could look at?

**Comment by mapledan** [comment]
These are binary and core-dump.
[coredump.zip](https://www.dropbox.com/s/158ofedqdh9rgcj/coredump.zip
)

**Comment by slact** [comment]
fixed in 73a3e259fce1bcd11f6c. It was a problem with the fix for #345. I've tagged release 1.1.6 with this fix.

**Comment by mapledan** [comment]
It's work. Thanks.
